internal/cli/flags.go

```diff
@@ -125,6 +134,10 @@ func registerBool(cmd *cobra.Command, f *Flag, value *bool, defaultValue bool, i
 
 func shouldAsk(cmd *cobra.Command, f *Flag, isUpdate bool) bool {
 	if isUpdate {
+		if !f.IsRequired {
+			return false
+		}
+
```

Comment:
@Widcket @jfatta btw this was a subtle bug -- TL;DR if it's `AskU` / `RegisterU` and not required, we should prompt for it. The assumption is most of our endpoints do `PATCH` semantics so for instance updating a rule shouldn't require the user to also update the name.

Reply:
I think [this change](https://github.com/auth0/auth0-cli/blob/170d9e59fe1346396c2cbd30f548e9f6b592584a/internal/cli/input.go#L18) fixes it.

Reply:
<img width="791" alt="Screen Shot 2021-03-20 at 11 00 19 AM" src="https://user-images.githubusercontent.com/2257/111880944-786f5400-896b-11eb-80d5-92038c7c55c3.png">

So when I undo this change even rebased on main I still get prompted for the Name. I think despite us passing `isRequired` in `TextInput` -- we'll still end up prompting for the Name -- which isn't the intent, correct?

Reply:
That's correct, if it is not a required input it should not prompt for it on update.

Reply:
Good catch!

Reply:
With this change `apps update` no longer prompts for callback URL. We should add a property for always prompting for that use case then.